### PR TITLE
Build all Erlang versions and upload to PackageCloud

### PR DESCRIPTION
**Pull request by carlhoerberg**
In a script, run by GH Actions.

### CODE REVIEW

File: .github/workflows/build-all-and-upload.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  build-all-and-upload:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.3'
```
Comment (dentarg) [inline .github/workflows/build-all-and-upload.yml:19]: This will always use the latest stable CRuby

```suggestion
          ruby-version: ruby
```

**Comment by dentarg** [comment]
An alternative approach could be like we have done over at [cloudamqp/rabbitmq-erlang](https://github.com/cloudamqp/rabbitmq-erlang/blob/4a7c65bf8505ce0bb70e95019345f9b6415774a8/.github/workflows/docker_http_trigger.yml#L3) where we have a workflow that can build a single version (in that case there, it is a version combo) and then we trigger multiple builds of that workflow. That would avoid long running builds, and make it more clear what version a certain build built, and easy to re-run individual builds.

**Comment by carlhoerberg** [comment]
Yes, was thinking of that too but couldn't figure out a good way to trigger new jobs, only [repository_dispatch](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#repository_dispatch). Depot.dev is also pretty fast and supports native arm64 builds.

**Comment by dentarg** [comment]
It would all live inside this repo so `workflow_dispatch` ought to work and the CLI could be used: https://cli.github.com/manual/gh_workflow_run (we would need to add a GitHub access token to the secrets, can't start new workflows with the default token). We could still make use of depot.dev.

**Comment by carlhoerberg** [comment]
ok! can continue tmrw

**Comment by carlhoerberg** [comment]
one workflow dispatch per image/erlang_version/platform? downside with that is that we will trigger a lot of workflows. alterantive is to use a matrix, so a single workflow but many jobs, but don't get how you can pass a hash to workflow_dispatch which then can be used as a `matrix: include`, if it was just an array i guess it would work. Maybe that's a comprimise, that you can only specify a version, and then we build that version for all platforms and images regardless? Another upside of that is that we can limit the concurrency of the matrix/jobs.

**Comment by carlhoerberg** [comment]
can't limit the number of concurrent workflows afaik, and then we can run into issues with depot.dev..

**Comment by dentarg** [comment]
It is a bit quirky but our workflow https://github.com/cloudamqp/rabbitmq-erlang/blob/4a7c65bf8505ce0bb70e95019345f9b6415774a8/.github/workflows/docker_http_trigger.yml#L3-L20 takes an array of hashes as input

**Comment by dentarg** [comment]
We trigger that workflow from https://github.com/84codes/account-console/blob/bfa3775d3b7eaad0ff85bd9a22d07ca9271229e8/cloudamqp/bin/trigger-rabbitmq-container-builds#L109-L113 + https://github.com/84codes/account-console/blob/bfa3775d3b7eaad0ff85bd9a22d07ca9271229e8/cloudamqp/lib/tools/rabbitmq_container_releases.rb#L68-L87

**Comment by carlhoerberg** [comment]
Ok, thanks!

**Comment by carlhoerberg** [comment]
Used a multi step workflow instead, and `fromJson`.

**Pull request merged**